api/api/error_handler.py

```diff
@@ -55,6 +55,43 @@ def _cleanup_detail_field(detail: str) -> str:
     return ' '.join(str(detail).replace("\n\n", ". ").replace("\n", "").split())
 
 
+async def handle_expect_header(request: ConnexionRequest, exc: ContentSizeExceeded = None) -> ConnexionResponse:
+    """Handler for the 'Expect' HTTP header.
+    
+    Parameters
+    ----------
+    request : ConnexionRequest
+        Incoming request.
+    exc : ContentSizeExceeded
+        Raised exception if content size exceeds the limit.
+
+    Returns
+    -------
+    Response
+        HTTP Response returned to the client.
+    """
+
+    if 'Expect' in request.headers:
+        expect_value = request.headers["Expect"].lower()
+
+        continue_msg = '100-continue'
+        data = continue_msg
+        status_code = 200
+        content_type = None
+
+        if expect_value != continue_msg or (exc and expect_value == continue_msg):
```

Comment:
This error handler cannot be the one checking whether the `Expect` header is correct. Notice that this function will only be called when a 417 error is raised in another part of the framework. Therefore, if the header value is being checked here, it means that it is not being checked elsewhere and as a consequence, 417 will not be raised and this function will not be called. 

In fact, this is not working right now:
```
# curl -k -X GET "https://localhost:55000/?pretty=true" -H "Authorization: Bearer $TOKEN" -H "Expect: Test_header"
{
   "data": {
      "title": "Wazuh API REST",
      "api_version": "4.9.0",
      "revision": 40900,
      "license_name": "GPL 2.0",
      "license_url": "https://github.com/wazuh/wazuh/blob/v4.9.0/LICENSE",
      "hostname": "wazuh-master",
      "timestamp": "2024-03-21T12:10:37Z"
   },
   "error": 0
}
```

We should create a new middleware which checks if the `Expect` header is included, if `100-continue` is contained, and probably, in case a `Content-length` header was also included, whether its value is bigger than the one defined in `api_conf['max_upload_size']`. This way, clients will ask/expect the API whether a payload is too big. If the API responds 417, it means that said content-length should not be sent because it would be refused.

In summary, the new middleware should raise a 417 error if any of the above mentioned conditions are unmet. Then, the API will catch the 417 error (`app.add_error_handler(417, error_handler.expect_header_error_handler)` in `wazuh-apid.py`) and will call this handler, which should format the response as it is being done below.